### PR TITLE
fix(eventing): add init_ext to use spawner arg and upgrade async_nats

### PR DESCRIPTION
**Pull request by datacore-vvarakantham**
Init_ext function in event handler to pass task spawner as a param to run the eventing publisher for io-engine. 
Upgraded async_nats to 0.31.0.

### CODE REVIEW

File: events-api/src/common/retry.rs [hunk 1]
```diff
@@ -60,7 +60,7 @@ impl BackoffOptions {
     }
 
     /// Specify a max number of retries before giving up.
-    pub fn with_max_retries(mut self, max_retries: u32) -> Self {
+    pub fn _with_max_retries(mut self, max_retries: u32) -> Self {
```
Comment (tiagolobocastro) [inline events-api/src/common/retry.rs]: why the the `_` ?

File: events-api/src/mbus_nats.rs [hunk 2]
```diff
@@ -265,23 +257,29 @@ impl Bus for NatsMessageBus {
...
-            if let Some(error) = err.downcast_ref::<std::io::Error>() {
-                if error.kind() == ErrorKind::TimedOut {
+
+            match err.kind() {
+                PublishErrorKind::TimedOut => {
                     tries += 1;
                     continue;
                 }
+                PublishErrorKind::StreamNotFound => {
+                    let _ = self.get_or_create_stream(None, None).await?;
```
Comment (tiagolobocastro) [inline events-api/src/mbus_nats.rs:277]: Should this yield tries++ as well?

File: events-api/src/mbus_nats.rs [hunk 3]
```diff
@@ -320,7 +318,12 @@ impl<T: Serialize + DeserializeOwned> BusSubscription<T> {
                 let message = match message {
                     Ok(message) => message,
                     Err(error) => {
-                        tracing::warn!(%error, "Error accessing jetstream message");
+                        match error.kind() {
+                            MessagesErrorKind::ConsumerDeleted => (), // TODO create consumer again
+                            MessagesErrorKind::MissingHeartbeat => (), /* TODO check if stream */
+                            // and consumer exists,
+                            _ => tracing::warn!(%error, "Error accessing jetstream message"),
+                        };
```
Comment (tiagolobocastro) [inline events-api/src/mbus_nats.rs]: Don't mix comment types.

```suggestion
                        match error.kind() {
                            // TODO: create consumer again.
                            MessagesErrorKind::ConsumerDeleted => (), 
                            // TODO: check if stream and consumer exists.
                            MessagesErrorKind::MissingHeartbeat => (),
                            _ => tracing::warn!(%error, "Error accessing jetstream message"),
                        };
```

File: events-api/src/common/retry.rs (hunk 1 above)
Comment (datacore-vvarakantham) [reply events-api/src/common/retry.rs]: The function with_max_retries is currently unused. It was previously used by the verify_stream_exists function, but it is removed now . So, I added  _.

Comment (tiagolobocastro) [reply events-api/src/common/retry.rs]: this function is pub, so how do you know it's unused by a client library?
If you don't want it to be used them remove it completely please.

Comment (datacore-vvarakantham) [reply events-api/src/common/retry.rs]: Removed the function now. 

/// Retry module for jetstream server.
pub(crate) mod retry;
This fn was in retry module which is used by only the library.

File: events-api/src/mbus_nats.rs (hunk 2 above)
Comment (datacore-vvarakantham) [reply events-api/src/mbus_nats.rs:277]: Yes. Added now thanks.

File: events-api/src/event_traits.rs [hunk 4]
```diff
@@ -13,14 +13,22 @@ pub fn initilize_source_component(comp: &str) {
 
 impl EventMeta {
     /// New event metadata with default values.
-    pub fn new() -> Self {
+    fn new() -> Self {
```
Comment (datacore-vvarakantham) [inline events-api/src/event_traits.rs:16]: Removed pub from new as all the event generation fns can now use from_source to get event meta.

**Review by tiagolobocastro — APPROVED** [review]

**Review by sinhaashish — APPROVED** [review]

**Pull request merged**